spikewidgets/widgets/collisioncompwidget/collisioncompwidget.py

```diff
@@ -0,0 +1,238 @@
+import numpy as np
+import sklearn
+
+from matplotlib import pyplot as plt
+import matplotlib.gridspec as gridspec
+
+
+from spikewidgets.widgets.basewidget import BaseWidget, BaseMultiWidget
+
+
+
+def plot_comparison_collision_pair_by_pair(comp, unit_ids=None, nbins=10, figure=None, ax=None):
+    """
+    Plots multi sorting comparison graph.
+
+    Parameters
+    ----------
+    comp: CollisionGTComparison
+        The collision ground truth comparison object
+    unit_ids: list
+        List of considered units
+    nbins: int
+        Number of bins
+    figure: matplotlib figure
+        The figure to be used. If not given a figure is created
+    ax: matplotlib axis
+        The axis to be used. If not given an axis is created
+
+    Returns
+    -------
+    W: MultiCompGraphWidget
+        The output widget
+    """
+    W = ComparisonCollisionPairByPairWidget(
+        comp=comp,
+        unit_ids=unit_ids,
+        nbins=nbins,
+        figure=figure,
+        ax=ax
+    )
+    W.plot()
+    return W
+
+
+def plot_comparison_collision_by_similarity(comp, templates, metric='cosine_similarity', 
+        unit_ids=None, nbins=10, figure=None, ax=None):
+    
+    W = ComparisonCollisionBySimilarityWidget(
+        comp=comp,
+        templates=templates,
+        metric=metric,
+        unit_ids=unit_ids,
+        nbins=nbins,
+        figure=figure,
+        ax=ax
+    )
+    W.plot()
+    return W
+
+
+
+class ComparisonCollisionPairByPairWidget(BaseWidget):
```

Comment:
You should use `BaseMultiWidget` instead. It already takes care of all the gridspec/subplots stuff

Reply:
I known but I was happy with it... don't remember why.